src/Aspire.Dashboard/Model/ResourceViewModel.cs

```diff
@@ -281,12 +286,15 @@ public UrlViewModel(string name, Uri url, bool isInternal)
     }
 }
 
-public sealed record class VolumeViewModel(string? Source, string Target, string MountType, bool IsReadOnly) : IPropertyGridItem
+public sealed record class VolumeViewModel(int index, string Source, string Target, string MountType, bool IsReadOnly) : IPropertyGridItem
 {
-    string? IPropertyGridItem.Name => Source;
-
+    string IPropertyGridItem.Name => Source;
     string? IPropertyGridItem.Value => Target;
 
+    // Source could be empty for an anomymous volume so it can't be used as a key.
+    // Because there is no good key in data, use index of the volume in results.
+    object IPropertyGridItem.Key => index;
```

Comment:
Confirmed that using source is a problem when there are multiple anonymous volumes:

![image](https://github.com/user-attachments/assets/8efaab44-208e-4924-ba9a-fedcaef38a78)

I looked at the data turned from DCP today, and the k8s API spec - https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#volumemount-v1-core - and there doesn't seem to be a better id.